api/grpcserver/smesher_service.go

```diff
@@ -26,6 +26,7 @@ type SmesherService struct {
 	smeshingProvider  api.SmeshingAPI
 
 	streamInterval time.Duration
+	postOpts       activation.PostSetupOpts
```

Comment:
Instead of passing in the default options here, I think we should give smapp the option to set `ComputeBatchSize`. This parameter can make a big difference in initialization time on different machines and only fall back to `config.DefaultComputeBatchSize` if not provided by smapp.

Reply:
Yes and no. Smapp IMHO can already set the value as they need. The problem is that we would rather need https://github.com/spacemeshos/post/issues/125 than ask the user for the BatchSize. POST is already waaaay too complex with waaaay too many variables. I'd honestly rather leave an open possibility for power-users and hide it from normal users.

Reply:
We don't have to do this now. We can also just set `ComputeBatchSize` to `config.DefaultComputeBatchSize` (as commented below) and maybe do this in the future, similar how `Scrypt` options are handled...

Reply:
@piersy  important note, you somewhat cannot easily modify the batchSize after the process has started.

It's more like all-or-nothing approach here.

The best would be imho "defaults unless specified" in single call.

@fasmat scrypt options are "hardcoded" no one should modify them in easy way, I would not put them to the same bucket.

Reply:
I know it doesn't make sense to allow them to be able to be changed, but we are not treating them differently from any other post options besides not being able to change them via grpc.

Reply:
> The best would be imho "defaults unless specified" in single call

@pigmej Is that one of the 2 reasonable approaches I listed?

Reply:
>I know it doesn't make sense to allow them to be able to be changed, but we are not treating them differently from any other post options besides not being able to change them via grpc.

So as I understand it both compute batch size and scrypt options are currently user configurable, but only with the config provided at startup, if they shouldn't be then this becomes pretty simple since all options would be specified by the API call.

But why are they user configurable if they shouldn't be?

Reply:
> But why are they user configurable if they shouldn't be?

Because of how we handle our `config` packages 😅: https://github.com/spacemeshos/go-spacemesh/pull/4297#issuecomment-1508803081

Reply:
So it sounds like we need a construct similar to ethereum's chain config that allows setting network wide parameters, do we have such a thing?

But for the purposes of this PR I think the current approach is fine. Then we'll wan't a followup PR to remove the ability to set scrypt params per node. And it seems like we need a discussion around batch size.